pangea-sdk/service/ai_guard/api.go

```diff
@@ -22,55 +21,47 @@ func (e *aiGuard) GuardText(ctx context.Context, input *TextGuardRequest) (*pang
 	return request.DoPost(ctx, e.Client, "v1beta/text/guard", input, &TextGuardResult{})
 }
 
-type TextGuardArtifact struct {
-	Defanged bool   `json:"defanged"`
-	End      int    `json:"end"`
-	Start    int    `json:"start"`
-	Type     string `json:"type"`
-	Value    string `json:"value"`
-	Verdict  string `json:"verdict,omitempty"`
+type AnalyzerResponse struct {
+	Analyzer   string  `json:"analyzer"`
+	Confidence float64 `json:"confidence"`
 }
 
-type TextGuardSecurityIssues struct {
-	CompromisedEmailAddresses int `json:"compromised_email_addresses"`
-	MaliciousDomainCount      int `json:"malicious_domain_count"`
-	MaliciousIPCount          int `json:"malicious_ip_count"`
-	MaliciousURLCount         int `json:"malicious_url_count"`
-	RedactRuleMatchCount      int `json:"redact_rule_match_count"`
+type PromptInjectionResult struct {
+	AnalyzerResponses []AnalyzerResponse `json:"analyzer_responses"`
 }
 
-type TextGuardFindings struct {
-	ArtifactCount  int                     `json:"artifact_count"`
-	MaliciousCount int                     `json:"malicious_count"`
-	SecurityIssues TextGuardSecurityIssues `json:"security_issues"`
+type PiiEntity struct {
+	Type     string   `json:"type"`
+	Value    string   `json:"value"`
+	Redacted bool     `json:"redacted"`
+	StartPos *float64 `json:"start_pos,omitempty"`
 }
 
-type RedactRecognizerResult struct {
-	FieldType string `json:"field_type"` // The entity name.
-	Score     int    `json:"score"`      // The certainty score that the entity matches this specific snippet.
-	Text      string `json:"text"`       // The text snippet that matched.
-	Start     int    `json:"start"`      // The starting index of a snippet.
-	End       int    `json:"end"`        // The ending index of a snippet.
-	Redacted  bool   `json:"redacted"`   // Indicates if this rule was used to anonymize a text snippet.
+type PiiEntityResult struct {
+	Entities []PiiEntity `json:"entities"`
 }
 
-type RedactReport struct {
-	Count             int                      `json:"count"`
-	RecognizerResults []RedactRecognizerResult `json:"recognizer_results"`
+type MaliciousEntity struct {
+	Type     string                 `json:"type"`
+	Value    string                 `json:"value"`
+	Redacted *bool                  `json:"redacted,omitempty"`
```

Comment:
Does it make sense to be optional?

Reply:
It's marked as such in the OpenAPI spec 🤷